gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -59,21 +64,28 @@
   public static final Extract.TableType DEFAULT_TABLE_TYPE = Extract.TableType.APPEND_ONLY;
   public static final String DEFAULT_NAMESPACE_NAME = "KAFKA";
   public static final String ALL_TOPICS = "all";
+  public static final String NUM_OF_MAPPERS = "num.of.mappers";
+  public static final int DEFAULT_NUM_OF_MAPPERS = 100;
+  public static final String AVG_EVENT_SIZE = "avg.event.size";
+  public static final long DEFAULT_AVG_EVENT_SIZE = 1024;
 
   private final Set<String> moveToLatestTopics = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
   private final Map<KafkaPartition, Long> previousOffsets = Maps.newHashMap();
+  private final Map<KafkaPartition, Long> previousAvgSizes = Maps.newHashMap();
 
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
-    ImmutableList.Builder<WorkUnit> builder = new ImmutableList.Builder<WorkUnit>();
+    List<List<WorkUnit>> workUnits = Lists.newArrayList();
     Closer closer = Closer.create();
     try {
       KafkaWrapper kafkaWrapper = closer.register(KafkaWrapper.create(state));
       List<KafkaTopic> topics = getFilteredTopics(kafkaWrapper, state);
       for (KafkaTopic topic : topics) {
-        builder.addAll(getWorkUnitsForTopic(kafkaWrapper, topic, state));
+        workUnits.add(getWorkUnitsForTopic(kafkaWrapper, topic, state));
       }
-      return builder.build();
+      int numOfMultiWorkunits = state.getPropAsInt(NUM_OF_MAPPERS, DEFAULT_NUM_OF_MAPPERS);
```

Comment:
There's already a key named `mr.job.max.mappers` that defines the maximum number of mappers to launch. Can you use that?